src/test/scala/calc/RunTest.scala

```diff
@@ -38,4 +38,28 @@ class RunTest {
     assertRun(54.3, "54.3")
   }
 
+  @Test def runOpBinary(): Unit = {
+    assertRun(13.3, "12.3 + (23.12 - 12) / 11.12")
+    assertRun(12.2, "6.0 / 3 + 10.2")
+    assertRun(-215.68, "0.32 - (23.12 + 12.88) * 6")
+  }
+
+  @Test def runLetExpr(): Unit = {
+    assertRun(4.41, "let sum = 2.1 in sum * sum")
+    assertRun(113.0876, "let s = 2.3 in let t = 9.2323 in (t * 12) + s")
+  }
+
+  @Test def runLetException(): Unit = {
+   try{
+      assertRun(4.41, "let sum = 2.1 in s * sum")
+      fail()
+    } catch {
+      case e: Exception =>
```

Comment:
This would also catch the exception thrown by `fail()`, so this test is void; it always succeeds. One way to avoid this problem is the following:
```scala
assertTrue(Try {
  assertRun(4.41, "let sum = 2.1 in s * sum")
}.isFailure)
```